test/release/examples/primers/arrays.chpl

```diff
@@ -123,6 +123,18 @@ proc printArr(X: [?D] real) {
   writeln("after negating X within printArr, X is:\n", X, "\n");
 }
 
+//
+// Note that querying an array's domain via the ``.domain`` method or
+// the function argument query syntax does not result in a domain
+// expression that can be reassigned.  In particular, we cannot do:
```

Comment:
This could be slightly crisper / less mysterious by saying "results in a `const ref` domain expression, which cannot be reassigned."  

After the code example, you could also potentially go on to say "In the event that you need to re-assign an array's domain, you will need to name it and pass it by `ref` to any code that needs to modify it."  (to suggest the way forward for people who do want to re-assign domains).  Getting this to transition to the next bit cleanly might be slightly challenging, though it also feels convenient that that's wht's next (e.g., "Speaking of assigning domains...")

Reply:
> This could be slightly crisper / less mysterious by saying "results in a const ref domain expression, which cannot be reassigned."

Makes sense. Will do.

> After the code example, you could also potentially go on to say "In the event that you need to re-assign an array's domain, you will need to name it and pass it by ref to any code that needs to modify it." (to suggest the way forward for people who do want to re-assign domains). Getting this to transition to the next bit cleanly might be slightly challenging, though it also feels convenient that that's wht's next (e.g., "Speaking of assigning domains...")

I'll take a look but that transition does look difficult.

Slightly out of scope for this PR, but as a big fan of the domain concept in general, I am not super happy how late "Domains can also be declared and named" comes here, and how that sentence makes domains a secondary thing to arrays. But maybe the motivation for that is to have a "dive into Chapel" vibe in the primer and show how to create and use arrays in the quickest way possible. Tying that to the topic at hand: I think that's the reason that we'll have to leave somethings towards the later parts of the primer for now. 

One way to accept your suggestion break the Gordion Knot could be putting "Arrays are passed to functions by reference..." paragraph somewhere earlier and then make a slight change in "Domains can also be declared and named." to connect it better to the discussion here. Which still doesn't address my concern about domains coming in late in this primer, but probably will make things relatively smooth and informative.

Reply:
> But maybe the motivation for that is to have a "dive into Chapel" vibe in the primer and show how to create and use arrays in the quickest way possible.

I think that's probably the case.  "You know what arrays are, let's get you declaring some quickly.  OK, now let's tell you about how Chapel arrays are a bit different than what you may be used to."  Note that there is also a domains primer, which may or may not make you feel better.  That said, it only mentions their use in declaring arrays and leaves any demonstrations of that to this one.